src/main/java/io/r2dbc/postgresql/PostgresqlAuthenticationFailure.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.postgresql;
+
+import io.r2dbc.postgresql.message.backend.Field;
+
+import java.util.List;
+
+/**
+ * An exception that represents a authentication error.
+ */
+public class PostgresqlAuthenticationFailure extends PostgresqlServerErrorException {
+
+    /**
+     * Creates a new exception.
+     *
+     * @param fields the fields to be used to populate the exception
+     * @throws IllegalArgumentException if {@code fields} is {@code null}
+     */
+    public PostgresqlAuthenticationFailure(final List<Field> fields) {
```

Comment:
No need to make method args `final`.

Reply:
ok. I just got used to my company checkstyle 😂